Source/DataModels/Components/ComponentTransform.h

```diff
@@ -188,9 +196,42 @@ inline void ComponentTransform::SetScale(const float3& scale)
 {
 	sca = scale;
 
-	if (sca.x <= 0) sca.x = 0.0001f;
-	if (sca.y <= 0) sca.y = 0.0001f;
-	if (sca.z <= 0) sca.z = 0.0001f;
+	if (sca.x <= 0) {
+		sca.x = 0.0001f;
```

Comment:
```suggestion
std::max(sca.x, 0.0001f);
```
Same for other instances of this pattern. Though this is just personal preference, so feel free to ignore me :P